### PR TITLE
Require Grape 0.16.1 and fix route_xxx deprecations.

### PR DESCRIPTION
**Pull request by dblock**
This isn't ready to merge. there're issues with `route_params` and `route_hidden`.

Closes https://github.com/ruby-grape/grape-swagger/issues/364.

### CODE REVIEW

**Comment by dblock** [comment]
For now I am monkey-patching the deprecation to avoid too many warnings being spewed.

```ruby
module Grape
  class Router
    class Route
      private

      # TODO: remove when https://github.com/ruby-grape/grape-swagger/pull/368 and https://github.com/newrelic/rpm/pull/235 are finished
      def warn_route_methods(name, location, expected = nil)
      end
    end
  end
end
```

File: lib/grape-swagger/doc_methods.rb [hunk 1]
```diff
@@ -501,51 +504,57 @@ def setup(options)
...
+
             if operation[:parameters].any? { |param| param[:type] == 'File' }
               operation.merge!(consumes: ['multipart/form-data'])
             end
             operation.merge!(responseMessages: http_codes) unless http_codes.empty?
 
-            if route.route_entity
-              type = @@documentation_class.parse_entity_name(Array(route.route_entity).first)
-              if route.route_is_array
+            if route.entity
```
Comment (LeFnord) [inline lib/grape-swagger/doc_methods.rb]: please respect also `route.success` → see: [describing-methods](https://github.com/ruby-grape/grape#describing-methods)

Comment (dblock) [inline lib/grape-swagger/doc_methods.rb]: that needs a spec I suspect, wasn't caught

Comment (LeFnord) [inline lib/grape-swagger/doc_methods.rb]: `success` is the new `entity` and `failures` the new `http_codes`, think a small spec to proof both new work in the same way as the old one should be enough

File: .travis.yml [hunk 2]
```diff
@@ -10,14 +10,7 @@ rvm:
...
 env:
-  - GRAPE_VERSION=0.8.0
-  - GRAPE_VERSION=0.9.0
-  - GRAPE_VERSION=0.10.0
-  - GRAPE_VERSION=0.10.1
-  - GRAPE_VERSION=0.11.0
-  - GRAPE_VERSION=0.12.0
-  - GRAPE_VERSION=0.13.0
-  - GRAPE_VERSION=0.14.0
+  - GRAPE_VERSION=0.16.1
```
Comment (LeFnord) [inline .travis.yml]: could you add `GRAPE_VERSION=0.15.0`

**Comment by dblock** [comment]
@LeFnord once this is all fixed here, should I be merging it onto master or swagger-1.2?

**Comment by LeFnord** [comment]
@dblock merg to master, after this is done, I would release the new versions

**Comment by dblock** [comment]
Closing in favor of https://github.com/ruby-grape/grape-swagger/pull/375.

File: lib/grape-swagger/doc_methods.rb (hunk 1 above)
Comment (dblock) [inline lib/grape-swagger/doc_methods.rb]: Actually no need, these were declared as DSL sugar on top of the `entity` and `http_codes` implementation in https://github.com/ruby-grape/grape/blob/1097eb192a8601234b5465c23b1ab09101048792/lib/grape/dsl/desc.rb#L90, so they work today out of the box.

**Pull request closed without merging**